### ISSUE TITLE
Possible Bug in LocallyConnected2D Class

### ISSUE THREAD

**Issue by BrianPSA** [eclipse/deeplearning4j#6938]
#### Version Information
* Deeplearning4j version (Most libraries are versioned 1.0.0 Beta)
* platform information (OS, etc) Windows 10
* CUDA version, if used N/A
* NVIDIA driver version, if in use N/A

#### Issue Description

Please describe our issue, along with:
- expected behavior 
As far as I can tell from documentation and looking at class code I am setting up the layer correctly

- encountered behavior
"Exception in thread "main" org.nd4j.linalg.exception.ND4JIllegalStateException: Found null variable at index 5 (See full stack below)"
from the LocallyConnected2D layer in the model below.

There is an extra question also arising that dosn't make sense and may be contributing to this error message.
From the "H1" LocallyConnected2D layer I am geting a config print of "outputSize=[2, 3], **featureDim=8**)" which I don't understand the **featureDim=8** part.
I set the filter/nOut dimension to 2, therefore the featureDim should either be 2 or 12 (2*3 * 2)? What does 8 stand for?
I have also let the model self determine the input size for the next/output layer on from the H1 layer, and it's config print confirms its receiving 12 values from the H1 layer which makes sense to me as apposed to the 8 reported by H1 layer itself?.

/*-------------------------Model Code ----------------------------------------------------------------------*/
	private static void testComputationalGraph04() throws IOException {
		// Structural setup pattern "Create config then create model" 03
		//Initialize the user interface backend
//		UIServer uiServer = UIServer.getInstance();
		//Configure where the network information (gradients, score vs. time etc) is to be stored. Here: store in memory.
//		StatsStorage statsStorage = new InMemoryStatsStorage(); //Alternative: new FileStatsStorage(File), for saving and loading later
		//Attach the StatsStorage instance to the UI: this allows the contents of the StatsStorage to be visualized
//		uiServer.attach(statsStorage);
		
		int nEpochs = 1;//Number of epochs (full passes of training data) to train on
		String[] indef = new String[]{"I0"};
		String[] outdef = new String[]{"O0"};
		int batchSize = 10;
		int inputChannels = 2; //Lvl4
		int inputHeight = 3; //Lvl 3
		int inputWidth = 4; //Lvl 2
//		int inputDepth = 3;
		long seed = 6;
		Distribution dfltDist = new NormalDistribution(0, 0.1);
		int c1F = 32;  //Number of feature maps / channels / depth for each CNN layer
		
		
		ComputationGraphConfiguration cgmc = new NeuralNetConfiguration
		  .Builder()
			.seed(seed)
//			.dist(dfltDist)
			.weightInit(dfltDist)
//		    .weightInit(WeightInit.ONES)
			.updater(new Adam(0.01))
			.l2(0.0001)
			.activation(Activation.SOFTSIGN)
			//This is important so we can 'stack' the results later (Same/Strict/Truncate)
//			.convolutionMode(ConvolutionMode.Same)
			
		  .graphBuilder()
			.addInputs(indef)
			.setInputTypes(new InputType.InputTypeConvolutional(inputHeight,inputWidth,inputChannels))
			.addLayer("H1",
			  new LocallyConnected2D.Builder()
//			  new Convolution2D.Builder()
				.nIn(inputChannels) //from layer class this appears to be the channels size of input layer
				.setInputSize(inputHeight,inputWidth)
//				.padding(0,0)
				.kernelSize(2,2)
				.stride(1,1)
				.dilation(1,1)
				.convolutionMode(ConvolutionMode.Strict)
				.activation(Activation.SOFTSIGN) //Will use global activation set above if not set here
				.hasBias(true)
				.nOut(2)
				.build(),
			  indef[0])
			
			.addLayer(outdef[0],
			  new OutputLayer.Builder()
//				.nIn(2*3*1)
				.lossFunction(LossFunctions.LossFunction.MSE)
				.activation(Activation.SOFTSIGN)
//			    .activation(Activation.IDENTITY)
				.nOut(2)
				.build(),
			"H1")
			.setOutputs(outdef)
		  .build();
		
		ComputationGraph cgm = new ComputationGraph(cgmc);
		cgm.init();
		
		//print("H1", cgm.getLayer("H1").conf().toString());
		System.out.println("Locally Connected H1 W&B\n" + cgm.getLayer("H1").paramTable().toString());
		System.out.println("Locally Connected H1 Layer\n" + cgm.getLayer("H1").toString());
		System.out.println("Output O0 layer\n" + cgm.getLayer("O0").toString());
		CollectScoresIterationListener el1 = new CollectScoresIterationListener(100);
//		StatsListener el2 = new StatsListener(statsStorage);
		cgm.setListeners(el1);
		
		INDArray[] it = new NDArray[1];
		INDArray[] ot = new NDArray[1];
		INDArray[] ie = new NDArray[1];
		INDArray[] oee = new NDArray[1];;
		INDArray[] oep = new NDArray[1];;
		
		it[0] = Nd4j.create(new float[][][][]{{{{1,2,3,4},{5,6,7,8},{-1,-2,-3,-4}},{{1,2,3,4},{5,6,7,8},{-1,-2,-3,-4}}}});
		ot[0] = Nd4j.create(new float[][]{{26,26}});
		ie[0] = Nd4j.create(new float[][][][]{{{{1,3,2,4},{5,7,6,8},{-1,-3,-2,-4}},{{1,3,2,4},{5,7,6,8},{-1,-3,-2,-4}}}});
		oee[0] = Nd4j.create(new float[][]{{26,26}});

		MultiDataSet mt = new MultiDataSet(it, ot);
		MultiDataSet me = new MultiDataSet(ie, oee);
		DataSet de = new DataSet(ie[0],oee[0]);

		MultiDataSetIterator jt = new TestMultiDataSetIterator(1, mt);
		MultiDataSetIterator je = new SingletonMultiDataSetIterator(me);
		DataSetIterator jE = new SingletonDataSetIterator(de);

/*------------------------Outputs and Error Stack ----------------------------------------------------------*/

Locally Connected layer H1 Weights & Bias
{W=[[[    0.2873,    0.4408], 
  [   -0.0896,   -0.2076], 
  [    0.2364,   -0.4105], 
  [    0.4492,    0.1081], 
  [    0.0734,    0.0335], 
  [    0.0013,   -0.2676], 
  [   -0.1275,   -0.2017], 
  [   -0.4673,    0.2179]], 

 [[    0.2219,    0.1555], 
  [    0.0202,   -0.1480], 
  [    0.3318,    0.3151], 
  [   -0.1016,   -0.2782], 
  [    0.0780,    0.5569], 
  [   -0.2384,    0.0117], 
  [   -0.0061,   -0.0957], 
  [   -0.1543,   -0.5886]], 

 [[   -0.2855,   -0.6936], 
  [    0.6165,    0.2426], 
  [    0.2337,    0.1047], 
  [   -0.1845,    0.0976], 
  [    0.4774,    0.0649], 
  [    0.0225,   -0.2303], 
  [   -0.7341,   -0.2526], 
  [   -0.0526,   -0.0944]], 

 [[    0.3067,   -0.1861], 
  [    0.1711,    0.2975], 
  [   -0.0767,    0.2044], 
  [    0.0743,   -0.4396], 
  [   -0.5108,   -0.2384], 
  [   -0.2757,   -0.1017], 
  [    0.1264,    0.0395], 
  [   -0.2139,   -0.0133]], 

 [[    0.3425,    0.2230], 
  [   -0.6343,   -0.0334], 
  [    0.3761,   -0.2949], 
  [   -0.6632,    0.1108], 
  [   -0.1741,    0.2696], 
  [    0.5054,    0.4283], 
  [   -0.2773,   -0.0676], 
  [   -0.2197,   -0.2351]], 

 [[    0.2575,   -0.2131], 
  [    0.2222,    0.3648], 
  [    0.5683,   -0.0324], 
  [   -0.4626,   -0.3728], 
  [    0.8349,    0.2002], 
  [   -0.5430,   -0.2390], 
  [   -0.6554,   -0.1231], 
  [    0.3557,   -0.4191]]], b=[[         0,         0]]}
  
/*Locally Connected H1 Layer*/
AbstractLayer(input=null, preOutput=null, conf=NeuralNetConfiguration(layer=LocallyConnected2D(nIn=2, nOut=2, activation=SOFTSIGN, kernel=[2, 2], stride=[1, 1], padding=[0, 0], cm=Strict, dilation=[1, 1], hasBias=true, inputSize=[3, 4], outputSize=[2, 3], **featureDim=8**), miniBatch=true, maxNumLineSearchIterations=5, seed=6, optimizationAlgo=STOCHASTIC_GRADIENT_DESCENT, variables=[W, b], stepFunction=null, minimize=true, pretrain=false, cacheMode=NONE, iterationCount=0, epochCount=0), dropoutApplied=false, trainingListeners=[], index=1, maskArray=null, maskState=null, cacheMode=NONE, inputModificationAllowed=false, iterationCount=0, epochCount=0)

/*Output O0 layer*/
org.deeplearning4j.nn.layers.OutputLayer{conf=NeuralNetConfiguration(layer=OutputLayer(super=BaseOutputLayer(super=FeedForwardLayer(super=BaseLayer(activationFn=softsign, weightInit=DISTRIBUTION, biasInit=0.0, dist=NormalDistribution(mean=0.0, std=0.1), l1=0.0, l2=1.0E-4, l1Bias=0.0, l2Bias=0.0, iUpdater=Adam(learningRate=0.01, learningRateSchedule=null, beta1=0.9, beta2=0.999, epsilon=1.0E-8), biasUpdater=null, weightNoise=null, gradientNormalization=None, gradientNormalizationThreshold=1.0), **nIn=12**, nOut=2), lossFn=LossMSE(), hasBias=true, legacyBatchScaledL2=false)), miniBatch=true, maxNumLineSearchIterations=5, seed=6, optimizationAlgo=STOCHASTIC_GRADIENT_DESCENT, variables=[W, b], stepFunction=null, minimize=true, pretrain=false, cacheMode=NONE, iterationCount=0, epochCount=0), score=0.0, optimizer=null, listeners=[]}

/*Error Stack*/
Exception in thread "main" org.nd4j.linalg.exception.ND4JIllegalStateException: Found null variable at index 5
	at org.nd4j.autodiff.samediff.SameDiff.addArgsFor(SameDiff.java:1271)
	at org.nd4j.autodiff.functions.DifferentialFunction.<init>(DifferentialFunction.java:437)
	at org.nd4j.autodiff.functions.DifferentialFunction.<init>(DifferentialFunction.java:421)
	at org.nd4j.linalg.api.ops.DynamicCustomOp.<init>(DynamicCustomOp.java:77)
	at org.nd4j.linalg.api.ops.impl.shape.Concat.<init>(Concat.java:45)
	at org.nd4j.autodiff.functions.DifferentialFunctionFactory.concat(DifferentialFunctionFactory.java:1268)
	at org.nd4j.autodiff.samediff.SameDiff.concat(SameDiff.java:7985)
	at org.nd4j.autodiff.samediff.SameDiff.concat(SameDiff.java:7970)
	at org.deeplearning4j.nn.conf.layers.LocallyConnected2D.defineLayer(LocallyConnected2D.java:195)
	at org.deeplearning4j.nn.layers.samediff.SameDiffLayer.doInit(SameDiffLayer.java:221)
	at org.deeplearning4j.nn.layers.samediff.SameDiffLayer.activate(SameDiffLayer.java:79)
	at org.deeplearning4j.nn.graph.vertex.impl.LayerVertex.doForward(LayerVertex.java:110)
	at org.deeplearning4j.nn.graph.ComputationGraph.outputOfLayersDetached(ComputationGraph.java:2316)
	at org.deeplearning4j.nn.graph.ComputationGraph.output(ComputationGraph.java:1727)
	at org.deeplearning4j.nn.graph.ComputationGraph.output(ComputationGraph.java:1686)
	at org.deeplearning4j.nn.graph.ComputationGraph.output(ComputationGraph.java:1616)
	at main.java.attest.testClass.testComputationalGraph04(testClass.java:672)
	at main.java.attest.testClass.main(testClass.java:85)

Process finished with exit code 1

#### Contributing
If you'd like to help us fix the issue by contributing some code, but would
like guidance or help in doing so, please mention it!

**Comment by AlexDBlack** [comment]
> Most libraries are versioned 1.0.0 Beta

Please upgrade to 1.0.0-beta3 and try again - there were a ton of fixes in 1.0.0-beta2 and 1.0.0-beta3.
https://deeplearning4j.org/release-notes.html

Furthermore, make sure you aren't mixing and matching DL4J/ND4J/DataVec etc versions - they should all be the same (i.e., all should be 1.0.0-beta3) with no mixed versions.
